lib/thrift/generator/client.ex

```diff
@@ -0,0 +1,270 @@
+                message = "The server replied to #{mismatched_rpc_name}, but we sent #{rpc_name}"
+                exc = %TApplicationException{
+                  message: message,
+                  type: BinaryBase.exception_type(3)}
+                {:error, {:exception, exc}}
+
+              {:ok, {mismatched_sequence_id, _, ^rpc_name, _}} ->
+                message = "Invalid sequence id. The client sent #{sequence_id}, but the server replied with #{mismatched_sequence_id}"
+
+                exc = %TApplicationException{message: message,
+                                             type: BinaryBase.exception_type(4)
+                                            }
+                {:error, {:exception, exc}}
+
+              {:ok, {_mismatched_sequence_id, _mismatched_rpc_name, _}} ->
+                message = "Both sequence id and rpc name are wrong. The server is extremely uncompliant."
+                exc = %TApplicationException{message: message,
+                                             type: :sequence_id_and_rpc_name_mismatched}
+                {:error, {:exception, exc}}
+
+              {:error, _}=err ->
+                err
+            end
+          end
+
+          defp read_t_application_exception(
+                <<11::size(8),
+                1::16-unsigned,
+                message_size::32-signed,
+                message::binary-size(message_size),
+                rest::binary>>, accum) do
+            # read the message string
+            read_t_application_exception(rest, Map.put(accum, :message, message))
+          end
+          defp read_t_application_exception(
+                <<8::size(8),
+                2::16-unsigned,
+                type::32-signed,
+                rest::binary>>, accum) do
+            # read the type
+            exception_type = BinaryBase.exception_type(type)
+            read_t_application_exception(rest, Map.put(accum, :type, exception_type))
+          end
+          defp read_t_application_exception(<<0>>, accum) do
+            # read the field stop and return
+            accum
+          end
+        end
+      end
+    end
+
+    def generate_function(service_module, function) do
+
+      args_module = service_module
+      |> Module.concat(Service.service_module_name(function, :args))
+
+      response_module = Service.service_module_name(function, :response)
+
+      underscored_name = function.name
+      |> Atom.to_string
+      |> Macro.underscore
+      |> String.to_atom
+
+      bang_name = :"#{underscored_name}!"
+
+      vars = function.params
+      |> Enum.map(&Macro.var(&1.name, nil))
+
+      assignments = function.params
+      |> Enum.zip(vars)
+      |> Enum.map(fn {param, var} ->
+        quote do
+          {unquote(param.name), unquote(var)}
+        end
+      end)
+
+      rpc_name = Atom.to_string(function.name)
+
+      quote do
+        def unquote(underscored_name)(client, unquote_splicing(vars), timeout \\ 5000) do
```

Comment:
Ok, I'll do that. 
At the very least, it gives us flexibility in the future if we come up with better ideas. 

Good catch, btw.